spring/src/main/java/io/seata/spring/annotation/datasource/SeataAutoDataSourceProxyCreator.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.spring.annotation.datasource;
+
+import javax.sql.DataSource;
+import java.util.stream.Stream;
+
+import io.seata.rm.datasource.DataSourceProxy;
+import io.seata.spring.util.SpringProxyUtils;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aop.Advisor;
+import org.springframework.aop.TargetSource;
+import org.springframework.aop.framework.AdvisedSupport;
+import org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.beans.BeansException;
+
+/**
+ * @author xingfudeshi@gmail.com
+ */
+public class SeataAutoDataSourceProxyCreator extends AbstractAutoProxyCreator {
```

Comment:
I have an idea, ProxyCreator can be implement like this:
```
// define a mark interface
public interface SeataProxy {
}

// implements MethodInterceptor and IntroductionInfo
public class SeataAutoDataSourceProxyAdvice implements MethodInterceptor, IntroductionInfo {
    private final DataSourceProxy dataSourceProxy;

    public SeataAutoDataSourceProxyAdvice(DataSourceProxy dataSourceProxy) {
        this.dataSourceProxy = dataSourceProxy;
    }
    
    ...

    @Override
    public Class<?>[] getInterfaces() {
        return new Class[]{SeataProxy.class};
    }
}

// extends AbstractAutoProxyCreator
public class SeataAutoDataSourceProxyCreator extends AbstractAutoProxyCreator {
    protected abstract Object[] getAdvicesAndAdvisorsForBean(Class<?> beanClass, String beanName, TargetSource customTargetSource) throws BeansException {
        advice = new SeataAutoDataSourceProxyAdvice()
        // use DefaultIntroductionAdvisor
        advisor = new DefaultIntroductionAdvisor(advice);
        return new Object[]{advisor};
    }
 
    @Override    
    protected boolean shouldSkip(Class<?> beanClass, String beanName) {
        return SeataProxy.class.isAssignableFrom(beanClass) || beanName ....;
    }
}
```

Reply:
Good idea!!